### ISSUE TITLE
Add a flag or something similar to time Axis to allow  ISO 8601 timestamps

### ISSUE THREAD

**Issue by mkalte666** [epezent/implot#129]
Over here we usually format using either DD/MM/YYYY or using ISO 8601 (YYYY-MM-DD). It would be nice to be able to switch the formatting - i have not found a way to do so.

**Comment by epezent** [comment]
@mkalte666 and others -- is this something that you would ideally like to be able to switch to at runtime, or would a compile time option that swaps the position of day/month suffice?

**Comment by epezent** [comment]
We could expose a style variable flag such as this:

```cpp
// Date format options
enum ImPlotDateFmt_ {
    ImPlotDateFmt_MDY = 0, // 1/31/2020
    ImPlotDateFmt_DMY = 1, // 31/1/2020
    ImPlotDateFmt_YMD = 2, // 2020/1/31
    ImPlotDateFmt_ISO = 3  // 2020-01-31
};
```

Would this, combined with the current 24-hour clock option, sufficiently accommodate most users in non-US countries?

**Comment by mkalte666** [comment]
> @mkalte666 and others -- is this something that you would ideally like to be able to switch to at runtime, or would a compile time option that swaps the position of day/month suffice?

I think a runtime option would be better - ideally i would want my application to display the date according to the location it is running at (well, at least until the whole world finally starts using ISO dates anyway^^). 

> We could expose a style variable flag such as this:
> 
> [....]
> 
> Would this, combined with the current 24-hour clock option, sufficiently accommodate most users in non-US countries?

That would be sufficient, at least for my use. 

Thanks for your work, using implot has been a joy so far :)

**Comment by epezent** [comment]
I've added support for ISO 8601 formatted dates: https://github.com/epezent/implot/pull/132

It can be enabled with `ImPlotStyle::UseISO8601`.

![iso](https://user-images.githubusercontent.com/29577475/96072557-1f563e00-0e6a-11eb-885e-3c72a634c4a6.gif)

I decided against the four options I presented above -- it was going to be too messy with way we currently handle formatting. Inevitably, someone will ask for more formatting customization and we can rethink our approach then. It would obviously be great if we could let users specify the exact spec they want (see https://gist.github.com/mlconnor/1887156 for all possibilities). But for now, ISO 8601 should suffice for non-US users. 

Thanks again for the suggestion! If this suits your needs, go ahead and close the issue :)
